cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/MasterTaskExecutorService.java

```diff
@@ -47,7 +47,8 @@ final class MasterTaskExecutorService extends AbstractScheduledService {
 
   @Override
   protected ScheduledExecutorService executor() {
-    executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("master-env-executor"));
+    executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory(
+      "master-env-task-" + task.getName() + "-%d"));
```

Comment:
Since this is a single thread executor, you don't need the `-%d` part.

Reply:
I see, the underlying ThreadFactory always expects such postfix int format. I can remove this, the extra arg would be safely ignored.